Documentation/ceph-csi-drivers.md

```diff
@@ -0,0 +1,393 @@
+
+### Deploy RBACs for node plugins, csi-attacher and csi-provisioner (common to Cephfs and RBD):
+
+``` 
+kubectl create -f csi-attacher-rbac.yaml
+kubectl create -f csi-provisioner-rbac.yaml
+kubectl create -f csi-nodeplugin-rbac.yaml
+```
+Those manifests deploy service accounts, cluster roles and cluster role bindings.
+### Deploy csi-attacher and csi-provisioner containers:
+##### Cephfs
+
+```
+kubectl create -f csi-cephfsplugin-attacher.yaml
+kubectl create -f csi-cephfsplugin-provisioner.yaml
+```
+##### RBD
+```
+kubectl create -f csi-rbdplugin-attacher.yaml
+kubectl create -f csi-rbdplugin-provisioner.yaml
+```
+Deploys stateful sets for external-attacher and external-provisioner sidecar containers.
+### Deploy the CSI driver:
+##### Cephfs
+```
+kubectl create -f csi-cephfsplugin.yaml
+```
+##### RBD
+```
+kubectl create -f csi-rbdplugin.yaml
+```
+Deploys a daemon set with two containers: CSI driver-registrar and the driver.
+### Verify the plugin, its attacher and provisioner are running:
+```
+# kubectl get all
+
+NAMESPACE   NAME                                       READY   STATUS      RESTARTS   AGE
+default     pod/csi-rbdplugin-975c4                    2/2     Running     0          132m
+default     pod/csi-rbdplugin-attacher-0               1/1     Running     0          132m
+default     pod/csi-rbdplugin-provisioner-0            1/1     Running     0          132m
+
+NAMESPACE   NAME                                TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)         AGE
+default     service/csi-rbdplugin-attacher      ClusterIP   10.107.65.227    <none>        12345/TCP       132m
+default     service/csi-rbdplugin-provisioner   ClusterIP   10.108.120.159   <none>        12345/TCP       132m
+
+NAMESPACE   NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
+default     daemonset.apps/csi-rbdplugin     1         1         1       1            1           <none>          132m
+
+NAMESPACE   NAME                                         DESIRED   CURRENT   AGE
+default     statefulset.apps/csi-rbdplugin-attacher      1         1         132m
+default     statefulset.apps/csi-rbdplugin-provisioner   1         1         132m
+
+```
+You should see an output similar to this when you run `kubectl get all`.
+
+## Test the CSI driver
+
+Once the plugin is successfully deployed, test it by running the following example
+
+This example is based on the ceph-csi [examples](https://github.com/ceph/ceph-csi/tree/master/examples) directory. It will describe how to test the RBD csi-driver, but it is very similar to run the cephfs one.
+
```

Comment:
This storage class needs a `rbd` pool. Please mention that the pool can be created through rook's pool crd.